### ISSUE TITLE
version conflict with `google-cloud-documentai`

### ISSUE THREAD

**Issue by CircleOnCircles** [GoogleCloudPlatform/cloud-debug-python#29]
This is similar to #24 

>I found out that this dep google-python-cloud-debugger is incomppatible with google-cloud-documentai.

>jingdi is my internal library on private pypi, the point is that it depends very recent version of google-cloud-documentai.

>poetry is my dependency resolver.

```
❯ poetry add google-python-cloud-debugger
Using version ^2.16 for google-python-cloud-debugger

Updating dependencies
Resolving dependencies...

  SolverProblemError

      Because no versions of google-api-core match >1.22.2,<1.22.3 || >1.22.3,<1.22.4 || >1.22.4,<1.23.0 || >1.23.0,<1.24.0 || >1.24.0,<1.24.1 || >1.24.1,<1.25.0 || >1.25.0,<1.25.1 || >1.25.1,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev
   and google-api-core (1.22.2) depends on google-api-core (1.22.2), google-api-core (>=1.22.2,<1.22.3 || >1.22.3,<1.22.4 || >1.22.4,<1.23.0 || >1.23.0,<1.24.0 || >1.24.0,<1.24.1 || >1.24.1,<1.25.0 || >1.25.0,<1.25.1 || >1.25.1,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2).
      And because google-api-core (1.22.3) depends on google-api-core (1.22.3), google-api-core (>=1.22.2,<1.22.4 || >1.22.4,<1.23.0 || >1.23.0,<1.24.0 || >1.24.0,<1.24.1 || >1.24.1,<1.25.0 || >1.25.0,<1.25.1 || >1.25.1,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3).
      And because google-api-core (1.22.4) depends on google-api-core (1.22.4)
   and google-api-core (1.23.0) depends on google-api-core (1.23.0), google-api-core (>=1.22.2,<1.24.0 || >1.24.0,<1.24.1 || >1.24.1,<1.25.0 || >1.25.0,<1.25.1 || >1.25.1,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0).
      And because google-api-core (1.24.0) depends on google-api-core (1.24.0)
   and google-api-core (1.24.1) depends on google-api-core (1.24.1), google-api-core (>=1.22.2,<1.25.0 || >1.25.0,<1.25.1 || >1.25.1,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1).
      And because google-api-core (1.25.0) depends on google-api-core (1.25.0)
   and google-api-core (1.25.1) depends on google-api-core (1.25.1), google-api-core (>=1.22.2,<1.26.0 || >1.26.0,<1.26.1 || >1.26.1,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1 || 1.25.0 || 1.25.1).
      And because google-api-core (1.26.0) depends on google-api-core (1.26.0)
   and google-api-core (1.26.1) depends on google-api-core (1.26.1), google-api-core (>=1.22.2,<1.26.2 || >1.26.2,<1.26.3 || >1.26.3,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1 || 1.25.0 || 1.25.1 || 1.26.0 || 1.26.1).
  (1) So, because google-api-core (1.26.2) depends on google-api-core (1.26.2)
   and google-api-core (1.26.3) depends on google-api-core (1.26.3), google-api-core (>=1.22.2,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1 || 1.25.0 || 1.25.1 || 1.26.0 || 1.26.1 || 1.26.2 || 1.26.3).

      Because no versions of google-api-core match >1.22.0,<1.22.1 || >1.22.1,<1.22.2
   and google-api-core (1.22.0) depends on google-api-core (1.22.0), google-api-core (>=1.22.0,<1.22.1 || >1.22.1,<1.22.2) requires google-api-core (1.22.0).
      And because google-api-core (1.22.1) depends on google-api-core (1.22.1), google-api-core (>=1.22.0,<1.22.2) requires google-api-core (1.22.0 || 1.22.1).
      And because google-api-core (>=1.22.2,<2.0.0dev) requires google-api-core (1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1 || 1.25.0 || 1.25.1 || 1.26.0 || 1.26.1 || 1.26.2 || 1.26.3) (1), google-api-core (>=1.22.0,<2.0.0dev) requires google-api-core (1.22.0 || 1.22.1 || 1.22.2 || 1.22.3 || 1.22.4 || 1.23.0 || 1.24.0 || 1.24.1 || 1.25.0 || 1.25.1 || 1.26.0 || 1.26.1 || 1.26.2 || 1.26.3)
      And because google-python-cloud-debugger (2.16) depends on google-api-core (1.15.0), google-api-core (>=1.22.0,<2.0.0dev) is incompatible with google-python-cloud-debugger (2.16).
  (2) So, because no versions of google-python-cloud-debugger match >2.16,<3.0
   and google-cloud-documentai (0.3.0) depends on google-api-core (>=1.22.0,<2.0.0dev), google-python-cloud-debugger (>=2.16,<3.0) is incompatible with google-cloud-documentai (0.3.0).

      Because jingdi (0.3.12) depends on google-cloud-documentai (>=0.3.0,<0.4.0)
   and no versions of jingdi match >0.3.12,<0.3.13 || >0.3.13,<0.4.0, jingdi (>=0.3.12,<0.3.13 || >0.3.13,<0.4.0) requires google-cloud-documentai (>=0.3.0,<0.4.0).
      And because jingdi (0.3.13) depends on google-cloud-documentai (>=0.3.0,<0.4.0)
   and no versions of google-cloud-documentai match >0.3.0,<0.4.0, jingdi (>=0.3.12,<0.4.0) requires google-cloud-documentai (0.3.0).
      And because google-python-cloud-debugger (>=2.16,<3.0) is incompatible with google-cloud-documentai (0.3.0) (2), google-python-cloud-debugger (>=2.16,<3.0) is incompatible with jingdi (>=0.3.12,<0.4.0)
      So, because gebwai-backend depends on both jingdi (^0.3.12) and google-python-cloud-debugger (^2.16), version solving failed.

  at ~\.poetry\lib\poetry\puzzle\solver.py:241 in _solve
      237│             packages = result.packages
      238│         except OverrideNeeded as e:
      239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
      240│         except SolveFailure as e:
    → 241│             raise SolverProblemError(e)
      242│
      243│         results = dict(
      244│             depth_first_search(
      245│                 PackageNode(self._package, packages), aggregate_package_nodes
```

**Comment by mctavish** [comment]
Oops.  We'll catch that on the next release.

**Comment by mctavish** [comment]
Should be resolved in v2.17
